backend/services/prisonerProfileService.ts

```diff
@@ -51,9 +32,46 @@ export default ({
       soc: { ui_url: socUrl, enabled: socEnabled },
       useOfForce: { prisons: useOfForcePrisons, ui_url: useOfForceUrl },
     },
-    app: { displayRetentionLink, esweEnabled },
+    app: { displayRetentionLink, esweEnabled, neurodiversityEnabledPrisons },
   } = config
 
+  // TODO: Part of temporary measure to only allow restricted prisons access to neurodivergence data. If no prisons specified then assume all are allowed access.
+  const canViewNeurodivergenceSupportData = (agencyId) => {
+    let canViewNeurodivergenceData = true
+    if (neurodiversityEnabledPrisons.length) {
+      const enabledPrisons = (neurodiversityEnabledPrisons as any).split(',')
+
+      canViewNeurodivergenceData = Boolean(
+        enabledPrisons.some((prison) => {
+          return prison === agencyId
+        })
+      )
+    }
+    return canViewNeurodivergenceData
+  }
+
+  const needNeurodivergenceSupport = (divergenceData) => {
+    const hasSupportNeed = divergenceData && divergenceData[0] !== null
+    if (hasSupportNeed && divergenceData[0][0]?.neurodivergenceSupport) {
```

Comment:
The previous one was checking values in 1 array (userCaseLoads) against values in another (accelerated prisons), this one is using agencyId instead of userCaseLoads, hence the slight difference and shorter code.

Reply:
To clarify after our convo: We are now getting data from the Curious API directly, which contains an array of an array.